File: scala/src/main/scala/io/devorbit/sparklabs/batch/sales/dataset/OrdersPerCity.scala
package io.devorbit.sparklabs.batch.sales.dataset

import org.apache.spark.sql.functions.{col, count, desc}

object OrdersPerCity extends App {
  DatasetUtils.toDataFrame(DatasetUtils.orders)
    .join(DatasetUtils.toDataFrame(DatasetUtils.customers), Seq("customer_id"), "inner")
    .groupBy(col("city"))
    .agg(count("*").as("total_orders_per_city"))
    .orderBy(desc("total_orders_per_city"))
    .show(false)
}

File: r/sparklabs/batch/sales/sql/orders_per_city.R
source("r/sparklabs/batch/sales/sql/utils.R")

showDF(run_sql("
  SELECT c.city, COUNT(*) AS total_orders_per_city
  FROM orders o
  JOIN customers c ON o.customer_id = c.customer_id
  GROUP BY c.city
  ORDER BY total_orders_per_city DESC
"), truncate = FALSE)
